cmd/modulegen/main.go

```diff
@@ -0,0 +1,152 @@
+package main
```

Comment:
it would be nice to have more godocs for this file, will be a lot easier to review imo and a lot easier to understand this file for future newcomers